Sources/Identifier.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Identifier.swift
+//  Valet
+//
+//  Created by Dan Federman and Eric Muller on 9/16/17.
+//  Copyright © 2017 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+
+public struct Identifier: CustomStringConvertible {
+    
+    // MARK: Initialization
+    
+    public init?(nonEmpty string: String?) {
```

Comment:
For maximum schwiftiness, it's possible to create a kind-of dependent types that would enforce this `String` being nonempty. See https://github.com/Ben-G/Validated/blob/master/Validated/Validated.swift for an example if y'all are interested 🙃

Reply:
`Validated` is absurdly nifty. Thanks for the link! However, in this case I think we don't need to go that hard-core. The external-facing API is the same for our `Identifier`, so we could always make this more Schwifty down the line without doing a major version bump :)